### PR TITLE
[Feature][K8S Task] support node selector

### PR DESCRIPTION
**Pull request by Radeity**
## Purpose of the pull request
- This PR is a partial job of https://github.com/apache/dolphinscheduler/issues/13868.

## Brief change log
- Support same definition of `NodeSelectorRequirement` with https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/node-selector-requirement/

     image

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-node-selectors.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        value: 'NotIn',
+        label: 'NotIn'
+    },
+    {
+        value: 'Exists',
+        label: 'Exists'
+    },
+    {
+        value: 'DoesNotExist',
+        label: 'DoesNotExist'
```
Comment (caishunfeng) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-node-selectors.ts:107]: nip
```suggestion
        value: 'NotExist',
        label: 'NotExist'
```

**Review by caishunfeng — DISMISSED** [review]
LGTM overall

**Comment by caishunfeng** [comment]
It's a good feature for k8s task.
BTW, if ds deploy in k8s, `node selector`  is the same with  `worker group`. Maybe we can replace worker group with node selector in k8s.

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-node-selectors.ts (hunk 1 above)
Comment (Radeity) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-node-selectors.ts:107]: I've checked K8S doc :
> You can use the operator field to specify a logical operator for Kubernetes to use when interpreting the rules. You can use In, NotIn, Exists, DoesNotExist, Gt and Lt.

Do we have to modify it?

**Comment by Radeity** [comment]
> It's a good feature for k8s task. BTW, if ds deploy in k8s, `node selector` is the same with `worker group`. Maybe we can replace worker group with node selector in k8s.

Agree with you, i can help with that.

**Comment by Radeity** [comment]
Hi, @caishunfeng, @SbloodyS, can you help re-run CI?

**Comment by SbloodyS** [comment]
> Hi, @caishunfeng, @SbloodyS, can you help re-run CI?

Done.

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**